File: wheystore/wheystore.Web/Modules/SanPhamDB/ChiTietDonHang/ChiTietDonHangColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.ChiTietDonHang")]
[BasedOnRow(typeof(ChiTietDonHangRow), CheckNames = true)]
public class ChiTietDonHangColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int ChiTietDonHangId { get; set; }
    public int MaDh { get; set; }
    public int MaSp { get; set; }
    [EditLink]
    public string TenSp { get; set; }
    public string AnhSp { get; set; }
    public decimal DonGia { get; set; }
    public decimal GiamGia { get; set; }
    public int SoLuong { get; set; }
    public decimal ThanhTien { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/ChiTietDonHangColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ChiTietDonHangRow } from "./ChiTietDonHangRow";

export interface ChiTietDonHangColumns {
    ChiTietDonHangId: Column<ChiTietDonHangRow>;
    MaDh: Column<ChiTietDonHangRow>;
    MaSp: Column<ChiTietDonHangRow>;
    TenSp: Column<ChiTietDonHangRow>;
    AnhSp: Column<ChiTietDonHangRow>;
    DonGia: Column<ChiTietDonHangRow>;
    GiamGia: Column<ChiTietDonHangRow>;
    SoLuong: Column<ChiTietDonHangRow>;
    ThanhTien: Column<ChiTietDonHangRow>;
}

export class ChiTietDonHangColumns extends ColumnsBase<ChiTietDonHangRow> {
    static readonly columnsKey = 'SanPhamDB.ChiTietDonHang';
    static readonly Fields = fieldsProxy<ChiTietDonHangColumns>();
}